js/apps/events-page/services/dateUtilityService.js

```diff
@@ -0,0 +1,68 @@
+((app) => {
+	const dateUtilityService = () => {
+		const addDays = (dateOrString, daysToAdd) => {
+			const date = typeof dateOrString === 'string' ? new Date(dateOrString) : dateOrString;
+
+			if (typeof date !== 'object' || date === 'Invalid Date') {
+				return date;
+			}
+
+			date.setDate(date.getDate() + daysToAdd);
+
+			return date;
+		};
+
+		const formatSchedule = (eventStart, eventLength) => {
+			if (!eventStart || isNaN(Date.parse(eventStart))) {
+				return 'Bad start date format';
+			}
+
+			if (typeof eventLength !== 'number' || eventLength <= 0) {
+				return 'Bad event length format';
+			}
+
+			const eventStartDate = new Date(eventStart);
+			const eventEndDate = new Date(eventStart);
+			eventEndDate.setMinutes(eventStartDate.getMinutes() + eventLength);
```

Comment:
something makes me uneasy about this, but can't really figure out any suggestions. maybe a new variable, which is probably overkill

Reply:
I moved it to an extra variable.